subsys/nrf_security/src/drivers/cracen/cracenpsa/src/rsaes_oaep.c

```diff
@@ -212,14 +207,18 @@ int cracen_rsa_oaep_encrypt(const struct sxhashalg *hashalg, struct cracen_rsa_k
 	/* start encoding and request generation of the random seed */
 
 	/* pointer used to walk through the data block DB */
-	uint8_t *datab = workmem.datablockstart;
+	uint8_t *datab = workmem.datablock + digestsz;
 	size_t datablocksz = modulussz - digestsz - 1;
-	size_t paddingstrsz = datablocksz - digestsz - text->sz - 1;
+	size_t paddingstrsz = (modulussz - digestsz - 1) - (digestsz + text->sz + 1);
 
-	datab += paddingstrsz;
+	if (paddingstrsz < 0 || paddingstrsz > datablocksz) {
```

Comment:
The paddingstrsz is a size_t which is unsigned so the test < 0 here doesn't make sense.  You will have to manually check if datablocksz is greater than (digestsz + text->sz + 1) .

Reply:
True, and actually we already test text.sz earlier for length earlier so this test is redundant anyways. Removed it.